platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ProblemReportCreator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems.internal;
+
+import java.io.File;
+import java.util.List;
+
+public interface ProblemReportCreator {
+    /**
+     * Stores a new problem in a temporary file that will be added to the final report when #createProblem is called.
```

Comment:
Please make sure that that the text is valid  javadoc (I think the createProblem reference will not work).